### ISSUE TITLE
Questions: Memory

### ISSUE THREAD

**Issue by BJWiley233** [whatshap/whatshap#313]
Hi,
I am trying to run a large `phase` job.  300 CRAM files with average size of between 30-40 GB per CRAM file.  I also have a 42 million line VCF file of "germline variants".  My LSF job terminated with code 137 for presumably terminated by signal 9.  It could have been killed by the cluster for some reason but that might be due to limits of Whathap?  I gave it 128GB of memory.  Should Whatshap be able to handle these CRAM files and VCF?

Brian

**Comment by ptrebert** [comment]
Hi,
while I don't have an ad hoc estimate for the resources required to run the job you describe, job scheduling systems such as LSF or PBS usually generate some form of log file with more precise information why exactly the job was killed (i.e., usually because the exceed time or memory limits). In other words, it would be advisable to first get your hands on this type of info and check what the scheduler reported as reason for killing the job (if LSF did it).

+Peter

**Comment by tobiasmarschall** [comment]
Is each CRAM for one sample? So your VCF would be 300 samples times ~42M variants? In that case, it'll likely be best to run WhatsHap for each sample separately on a VCF that only has that one sample (i.e. pull the VCF apart and later remerge, eg with bcftools). That's a bit extra work, but will scale better in a cluster environment.

**Comment by BJWiley233** [comment]
It's a total of 42 million variants for the 300 samples.  The WGS CRAM files are extremely large and so maybe that is what it cannot run all at once.  I couldn't even run Chromosome 1-18 with all 300 samples at once and had to break those in batches of 3.
Running 1 at a time sounds good.  I can do/prefer that actually clinically.  I thought that phasing does better when there are more samples being phased at once.  Is this not necessarily a true statement for Whatshap because it's a read aware phaser?

**Comment by tobiasmarschall** [comment]
Indeed, WhatsHap is doing pure read-based phasing. So unless you use pedigree mode (`--ped`, for related samples) each sample is phased completely independently. So the results of running each sample independently will be identical.

**Comment by ptrebert** [comment]
Ok, I am not familiar with LSF, but the PBS system we have is more explicit in the stderr logs of the job (e.g., statements like this `PBS: job killed: walltime 17973 exceeded limit 17940` make it clear what the problem was that led to killing the job).
But anyway, processing the samples independently as suggested by Tobias is definitely the first step to go.
